core/amber/src/main/scala/edu/uci/ics/texera/web/service/ExecutionResultService.scala

```diff
@@ -264,6 +273,15 @@ class ExecutionResultService(
               }
 
               if (StorageConfig.resultStorageMode == ICEBERG && !hasSingleSnapshot) {
+                val layerName = physicalPlan.operators
+                  .filter(physicalOp =>
+                    physicalOp.id.logicalOpId == opId &&
+                      physicalOp.outputPorts.keys.forall(outputPortId => !outputPortId.internal)
+                  )
+                  .headOption match {
+                  case Some(physicalOp: PhysicalOp) => physicalOp.id.layerName
+                  case None                         => "main"
+                }
```

Comment:
I hope in a future PR, we can avoid processing layer name. Try add a TODO if that's the plan.

Reply:
That will be the next PR. Added a TODO.